### ISSUE TITLE
Union and intersection type inconsistent behaviour when multi-line

### ISSUE THREAD

**Issue by KSXGitHub** [dprint/dprint#192]
## Steps to reproduce

1. Clone https://github.com/KSXGitHub/sane-fmt/tree/dprint-issue-report-intersection-hanging.
2. Switch to `dprint-issue-report-intersection-hanging` branch.
3. Run [./reproduce.sh](https://github.com/KSXGitHub/sane-fmt/blob/dprint-issue-report-intersection-hanging/reproduce.sh)

## Expected behavior

There shouldn't be any error as [./valid-typescript-file.ts](https://github.com/KSXGitHub/sane-fmt/blob/dprint-issue-report-intersection-hanging/valid-typescript-file.ts) is correctly formatted.

## Actual behavior

There is an error. It prints out a different format of `MyIntersection`, which is not what I want.

## Formatting Rules

I use `dprint-typescript-plugin` only (without `dprint-core`).

The formatting rules in located in `fn modify` of [rules/mod.rs:10-18](https://github.com/KSXGitHub/sane-fmt/blob/dprint-issue-report-intersection-hanging/src/rules/mod.rs#L10-L18).

## About the repo

It is an application that I am working on. I find it convenient to use this repo as bug report, so I made a separate branch for it.

**Comment by dsherret** [comment]
@KSXGitHub is this correct?

**Input**

```ts
export type MyUnion =
  | { type: 0; value: number }
  | { type: 1; value: string }
  | { type: 2; value: symbol }
```

**Expected Output**

```ts
export type MyUnion = { type: 0; value: number } | { type: 1; value: string } |
  { type: 2; value: symbol };
```

**Actual Output**

```ts
export type MyUnion =
  | { type: 0; value: number }
  | { type: 1; value: string }
  | { type: 2; value: symbol }
```

Right now once something is "multi-line" it will keep it that way. If you don't want that behaviour then `union_and_intersection_type_prefer_single_line` should be true as well, but that hasn't been implemented yet.

**Comment by dsherret** [comment]
Relates to #189. 

I think perhaps `preferSingleLine` should have a different name, but it will allow the behaviour you want. Also, will need to implement "operatorPosition" for union and intersection types.

**Comment by KSXGitHub** [comment]
You misunderstood. It was not the union that is the problem, it was the intersection. The TS file is already formatted as I desired, but dprint disagrees.

**Comment by dsherret** [comment]
@KSXGitHub could you provide the input, expected output, and actual output in this thread to help me understand? Thanks!

**Comment by KSXGitHub** [comment]
https://github.com/KSXGitHub/sane-fmt/blob/dprint-issue-report-intersection-hanging/valid-typescript-file.ts IS the expected output.

**Comment by dsherret** [comment]
Ok, so is this correct?

**Input**

```ts
export type MyIntersection =
  & { a: number }
  & { b: string }
  & { c: symbol }
```

**Expected Output**

```ts
export type MyIntersection =
  & { a: number }
  & { b: string }
  & { c: symbol }
```

**Actual Output**

```ts
export type MyIntersection = { a: number }
    & { b: string }
    & { c: symbol };
```

**Comment by KSXGitHub** [comment]
Yes.

**Comment by dsherret** [comment]
So this was actually done on purpose to copy what prettier does in some scenarios. For example:

![image](https://user-images.githubusercontent.com/1609021/80814886-b2d8d280-8b9a-11ea-9ac2-94b3342f57db.png)

That said, I think you're right and don't think it's good to have inconsistent behaviour here. I'm going to update them to work the same way.

**Comment by KSXGitHub** [comment]
Can you notify me when you release a new version with this fix?

**Comment by dsherret** [comment]
@KSXGitHub yeah, for sure. I'll do it after I close this issue. Just working on something else briefly for a bit, but I'll do it today :)

**Comment by dsherret** [comment]
@KSXGitHub published in 0.15.0.
